src/moon.c

```diff
@@ -0,0 +1,225 @@
+#include <predict/predict.h>
+#include "unsorted.h"
+#include <stdlib.h>
+#include <string.h>
+#include "defs.h"
+#include "sun.h"
+
+/**
+ * This function reduces angles greater than two pi by subtracting two pi
+ * from the angle
+ *
+ * \copyright GPLv2+
+ **/
+double FixAngle(double x)
+{
+
+	while ( x > 2*M_PI )
+		x-=2*M_PI;
+
+	return x;
+}
+
+/**
+ * This function is used in the FindMoon() function.
+ *
+ * \copyright GPLv2+
+ **/
+double PrimeAngle(double x)
+{
+	x=x-360.0*floor(x/360.0);
+	return x;
+}
+
+/* This function determines the position of the moon, including the azimuth and elevation headings, relative to the latitude
+	   and longitude of the tracking station.  This code was derived
+	   from a Javascript implementation of the Meeus method for
+	   determining the exact position of the Moon found at:
+	   http://www.geocities.com/s_perona/ingles/poslun.htm. */
+void predict_observe_moon(const predict_observer_t *observer, double time, struct predict_observation *obs)
+{
+	
+	double	jd, ss, t, t1, t2, t3, d, ff, l1, m, m1, ex, om, l,
```

Comment:
It's common these days to have one declaration on each line, i.e.,

double jd;
double ss;
...

and to declare the variables as late as possible instead of at the beginning of a block. There are several reasons. It puts the declaration closer to where it's used, so one doesn't have to go hunting for the declaration. It also makes it possible to combine the declaration and initial assignment, and makes it easier to check that one hasn't forgot to define a variable. E.g., in this case, it's hard to know if we've remembered to define all of these, or if we forgot one.

Not saying you have to change this now, though. Especially since you're just moving old code. Just advice for future patches.